test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -1659,6 +1663,51 @@ public void indexRandom(boolean forceRefresh, boolean dummyDocuments, boolean ma
         }
     }
 
+    /*
+    * This method ingests bogus documents for the given indices such that multiple slices
+    * are formed. This is greatly useful for testing with the concurrent search use-case.
+    * @param refreshCount    the number of segments to be formed by ingesting bogus documents
+    * @param indices         the indices in which bogus documents should be ingested
+    * */
+    public Set<List<String>> indexRandomForMultipleSlices(int segments, String... indices) throws InterruptedException {
```

Comment:
We probably shouldn't thrown `InterruptedException` but `RuntimeException` instead (it pops up in all test cases otherwise). Also, I think this method should be `protected`

Reply:
@reta I know what u are referring to but we have a latch usage in the function, hence its throwing `InterruptedException`. Let me know if you have any suggestions.

Also, I agree to make it protected, I will make that change.

Reply:
```
try {
 ...
} catch (InterruptedException ex) {
    Thread.currentThread().interrupt();
    throw new RuntimeException("The wait was interrupted", ex");
}
```

Reply:
indexRandom function is also throwing `InterruptedException` exception. Hence, I was thinking to do the same.

I know if we change it to RunTimeException, we will have to change the tests calling this method. But we are anyways making changes in the tests, this shouldn't be a big adhoc. What do u think?